webapp/src/views/Domain/CrudDomain.vue

```diff
@@ -0,0 +1,101 @@
+<script>
+import CrudList from '@/views/CrudList';
+
+export default {
+  name: 'CrudDomain',
+  extends: CrudList,
+  data: function () {
+    const self = this;
+    return {
+      createable: true,
+      updatable: true,
+      destroyable: true,
+      headlines: {
+        table: `Record Sets ${self.$route.params.domain}`,
+        create: `Create New Record Set (${self.$route.params.domain})`,
+        destroy: 'Delete Record Set',
+      },
+      texts: {
+        create: () => ('Create a record set'),
+        destroy: rrset => (`Delete record set ${rrset.type} ${rrset.subname}?`),
+        destroyInfo: () => ('This operation will permanently remove this information from the DNS.'),
+      },
+      columns: {
+        type: {
+          name: 'item.type',
+          text: 'Type',
+          textCreate: 'Record Set Type',
+          align: 'left',
+          sortable: true,
+          value: 'type',
+          readonly: false,
+          datatype: 'RRSetType',
+          searchable: true,
+        },
+        subname: {
+          name: 'item.subname',
+          text: 'Subname',
+          align: 'left',
+          sortable: true,
+          value: 'subname',
+          readonly: true,
+          datatype: 'GenericText',
+          searchable: true,
+          writeOnCreate: true,
+        },
+        records: {
+          name: 'item.records',
+          text: 'Content',
+          textCreate: 'Record Set Content',
+          align: 'left',
+          sortable: false,
+          value: 'records',
+          readonly: false,
+          datatype: 'RRSet',
+          fieldProps: rrSet => ({ type: rrSet.type || 'A' }),
+          searchable: true,
+        },
+        ttl: {
+          name: 'item.ttl',
+          text: 'TTL',
+          align: 'left',
+          sortable: true,
+          value: 'ttl',
+          readonly: false,
+          datatype: 'GenericText', // TODO TTL is not a String
+          fieldProps: () => ({ type: 'number' }),
+          searchable: true,
+        },
+        touched: {
+          name: 'item.touched',
+          text: 'Last touched',
+          align: 'left',
+          sortable: true,
+          value: 'touched',
+          readonly: true,
+          datatype: 'TimeAgo',
+          searchable: false,
+        },
+      },
+      actions: [
+      ],
+      paths: {
+        list: 'domains/::{domain}/rrsets/', // TODO dangerous?
```

Comment:
The domain value inserted here is part of the URL. Is it possible to craft an URL such that the user here triggers and authenticates unwanted API actions?

Reply:
As far as I can see, that is not possible: it only would be possible to construct other paths that end in `rrsets/`, but there are in fact no other such endpoints. If you try to insert something else so that the `rrsets/` in the end is in excess, our API gives 404.

I admit though that this is probably not the ideal solution. I also remember we had the same discussion when you initially wrote this, but we decided it's ok.